// Copyright 2016 Peter Goodman (peter@trailofbits.com), all rights reserved.

package remill.cfg;

message Instr {
    required    bytes       bytes = 1;
    required    uint64      address = 2;
}

message Block {
    required    uint64      address = 1;
    repeated    Instr       instructions = 2;
}

enum Visibility {
    EXPORTED    = 0;
    IMPORTED    = 1;
}

message NamedBlock {
    required    string      name = 1;
    required    uint64      address = 2;
    required    Visibility  visibility = 3;
}

message Module {
    repeated    Block       blocks = 1;
    repeated    NamedBlock  named_blocks = 2;
    repeated    uint64      referenced_blocks = 3;
    repeated    uint64      addressed_blocks = 4;
}
